pages/_app.js

```diff
@@ -1,109 +1,116 @@
 import Head from 'next/head'
 import fetch from 'isomorphic-unfetch'
-import { useRef, useState, useEffect } from 'react'
-import getConfig from 'next/config'
+import { useState, useEffect } from 'react'
 import {useRouter} from 'next/router'
-import AppContext from './AppContext'
 import CountrySelector from './Components/CountrySelector'
+import rootReducer from './reducers'
+import { createStore } from 'redux'
+import { Provider, useSelector, useDispatch } from 'react-redux'
+import { receiveCountries, selectCountry, selectYear, fetchCountries } from './actions'
 
-function App({ Component, pageProps }) {
+const App = () => {
```

Comment:
Also, for each component which is receiving the extra properties `props`, you need to define what properties are expected.

https://reactjs.org/docs/typechecking-with-proptypes.html

Reply:
added prop-types.